xarray/core/rolling_exp.py

```diff
@@ -128,9 +129,41 @@ def mean(self, keep_attrs: bool | None = None) -> T_DataWithCoords:
         if keep_attrs is None:
             keep_attrs = _get_keep_attrs(default=True)
 
-        return self.obj.reduce(
-            move_exp_nanmean, dim=self.dim, alpha=self.alpha, keep_attrs=keep_attrs
-        )
+        dim_order = self.obj.dims
+
+        from .dataset import Dataset
+
+        if isinstance(self.obj, Dataset):
+            new_ds = self.obj.copy()
+            if not keep_attrs:
+                new_ds.attrs = {}
+            for name, var in self.obj.variables.items():
+                if self.dim in self.obj[name].dims:
+                    new_ds[name] = apply_ufunc(
```

Comment:
This is required because `apply_ufunc` attempts to call the function on all objects, and doesn't exclude objects that don't have that `core_dim`.

Would it make sense to change `apply_ufunc` to skip those variables? Or provide a parameter to control this, or a convenience method?

Without this issue, also the `keep_attrs` boilerplate possibly goes away too.

Reply:
I'd be +1 (related #7027)

Reply:
See https://github.com/pydata/xarray/issues/6391

I agree that we should  just ignore variables without core  dims by default.